### PR TITLE
Api write categories

### PR DESCRIPTION
**Pull request by Azure7111**
https://community.openproject.org/work_packages/17415
https://community.openproject.org/work_packages/17414

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://feature-api-write-categories.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://feature-api-write-categories.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by Azure7111** [comment]
Also fixes a test failure from https://community.openproject.org/work_packages/17413

**Comment by ulferts** [comment]
@Azure7111 when testing the code I wasn't able to produce a validation error. So using an invalid href value for categories e.g. "category":{"href":"/api/v3/categories/-1"} does not trigger a validation error. Not for the POST form endpoint and not for the PATCH work_package/:id endpoint.

**Comment by ulferts** [comment]
@Azure7111 this needs rebasing, sorry.

File: lib/api/v3/work_packages/form/schema_allowed_categories_representer.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+            { href: api_v3_paths.category(category.id), title: category.name }
+          end
+
+          self.type = 'Category'
+        end
+      end
+    end
+  end
+end
```
Comment (ulferts) [inline lib/api/v3/work_packages/form/schema_allowed_categories_representer.rb]: This file can be deleted now that #2536 has been merged in.

**Comment by ulferts** [comment]
@Azure7111 good to merge once the now outdated file is removed.

**Pull request merged**